### ISSUE TITLE
CODEOWNERS validation does not enforce owner order

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#113]
## Description

The CODEOWNERS validation uses order-independent matching for owners. A rule with `@owner1 @owner2` in the file will pass validation even if the config specifies `["@owner2", "@owner1"]`.

## Why Order Matters

In GitHub CODEOWNERS:
- Owners are notified in the order they appear
- The first owner is typically considered the primary reviewer
- Some organizations use order to indicate primary vs secondary ownership

## Reproduction

CODEOWNERS file:
```
* @owner1 @owner2
```

check.toml:
```toml
[[process.codeowners.rules]]
pattern = "*"
owners = ["@owner2", "@owner1"]
```

Running `cm process check` passes, even though the order is different.

## Location

`src/process/tools/codeowners.ts` (lines 178-184)

```typescript
private ownersMatch(expected: string[], actual: string[]): boolean {
  if (expected.length !== actual.length) {
    return false;
  }
  const expectedSet = new Set(expected);  // Order lost here
  return actual.every((owner) => expectedSet.has(owner));
}
```

## Expected Behavior

Should enforce exact order match, or provide a config option `strict_order = true/false`.

## Suggested Fix

```typescript
private ownersMatch(expected: string[], actual: string[]): boolean {
  if (expected.length !== actual.length) {
    return false;
  }
  // Enforce order
  return expected.every((owner, index) => owner === actual[index]);
}
```

## Environment

- check-my-toolkit version: 1.3.0
